proto/minder/v1/minder.proto

```diff
@@ -793,7 +793,7 @@ message DeleteRepositoryByNameResponse {
 
 message ListRepositoriesRequest {
     string provider = 1 [deprecated=true];
-    int32 limit = 3;
+    int64 limit = 3;
```

Comment:
This /seems/ to be OK based on https://protobuf.dev/programming-guides/proto3/#updating which says "int32, uint32, int64, uint64, and bool are all compatible – this means you can change a field from one of these types to another without breaking forwards- or backwards-compatibility. If a number is parsed from the wire which doesn’t fit in the corresponding type, you will get the same effect as if you had cast the number to that type in C++ (for example, if a 64-bit number is read as an int32, it will be truncated to 32 bits)." but I want to add an explicit comment and leave the PR around for a little bit so that someone else with a better protobuf fu can take a look, too.

Reply:
I'll also add: https://buf.build/docs/breaking/rules#field_wire_json_compatible_type

Reply:
Hmm, we never invoke `buf breaking` in the CI https://buf.build/docs/reference/cli/buf/breaking. Let me check whether this is ok or not (will add to CI too)

https://protobuf.dev/programming-guides/proto3/#json:

> | proto3 | JSON | JSON example	 | Notes	 |
> |--------|--------|--------|--------|
> | int32, fixed32, uint32 | number | `1, -10, 0` | JSON value will be a decimal number. Either numbers or strings are accepted. |
> | int64, fixed64, uint64 | string | `"1", "-10"` | JSON value will be a decimal string. Either numbers or strings are accepted. |

Reply:
It fails _without_ `FIELD_WIRE_JSON_COMPATIBLE_TYPE` :

```
$ buf breaking proto/minder/v1/minder.proto --against '.git#branch=main'
proto/minder/v1/minder.proto:796:5:Field "3" on message "ListRepositoriesRequest" changed type from "int32" to "int64". See https://developers.google.com/protocol-buffers/docs/proto3#updating for wire compatibility rules and https://developers.google.com/protocol-buffers/docs/proto3#json for JSON compatibility rules.
```
So this is a breaking change.

Reply:
oh is it a breaking change because of the change in JSON? I was going based on the proto3-updating link which says "int32, uint32, int64, uint64, and bool are all compatible – this means you can change a field from one of these types to another without breaking forwards- or backwards-compatibility. " so I'm confused.

@evankanderson thoughts?

Reply:
It mentions on top (https://protobuf.dev/programming-guides/proto3/#updating):

> Note
> If you use JSON or [proto text format](https://protobuf.dev/reference/protobuf/textformat-spec) to store your protocol buffer messages, the changes that you can make in your proto definition are different.

But IMO the docs should be improved, it should define what is _different_.

Reply:
I believe @evankanderson enabled the `FIELD_WIRE_JSON_COMPATIBLE_TYPE` - he may thoughts on this thread.

Reply:
When converting from int32 to int64, the proto form of the _output_ will change from string to int.  It will accept both JSON numbers (IEEE 64-bit ints) and numeric strings as input.  Proto does this for JSON serialization because JSON numbers will lose precision around 2^53, and the proto team feels that is not acceptable.

Given that we've got other limits that probably mean limit should be < 2**16, I'm indifferent as to whether this is an int64 or stays an int32.